clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -109,6 +110,21 @@ static bool IsArtificial(VarDecl const *VD) {
                               cast<Decl>(VD->getDeclContext())->isImplicit());
 }
 
+static bool usesDebugTransparent(const Decl *D, const CodeGenModule &CGM) {
+  if (!D)
+    return false;
+
+  if (auto *attr = D->getAttr<DebugTransparentAttr>()) {
```

Comment:
```suggestion
  if (D->hasAttr<DebugTransparentAttr>()) {
```

Reply:
Might also be worth inverting this condition and preferring early-return style.

Reply:
I need the `attr` variable to access its location when reporting that it's missing.

Reply:
ah right, missed that